pkg/ui/react-app/src/thanos/pages/stores/StorePoolPanel.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { FC } from 'react';
+import { Container, Collapse, Table, Badge } from 'reactstrap';
+import { now } from 'moment';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faInfinity } from '@fortawesome/free-solid-svg-icons';
+import { ToggleMoreLess } from '../../../components/ToggleMoreLess';
+import { useLocalStorage } from '../../../hooks/useLocalStorage';
+import { getColor } from '../../../pages/targets/target';
+import { formatRelative, formatTime, parseTime } from '../../../utils';
+import { Store } from './store';
+import StoreLabels from './StoreLabels';
+
+export type StorePoolPanelProps = { title: string; storePool: Store[] };
+
+export const columns = [
+  'Endpoint',
+  'Status',
+  'Announced LabelSets',
+  'Min Time',
+  'Max Time',
+  'Last Successful Health Check',
+  'Last Message',
+];
+
+const MAX_TIME = 9223372036854775807;
+
+export const StorePoolPanel: FC<StorePoolPanelProps> = ({ title, storePool }) => {
+  const [{ expanded }, setOptions] = useLocalStorage(`store-pool-${title}-expanded`, { expanded: true });
+
+  return (
+    <Container fluid>
+      <ToggleMoreLess event={(): void => setOptions({ expanded: !expanded })} showMore={expanded}>
+        <span style={{ textTransform: 'capitalize' }}>{title}</span>
+      </ToggleMoreLess>
+      <Collapse isOpen={expanded}>
+        <Table size="sm" bordered hover>
+          <thead>
+            <tr key="header">
+              {columns.map(column => (
+                <th key={column}>{column}</th>
+              ))}
+            </tr>
+          </thead>
+          <tbody>
+            {storePool.map((store: Store) => {
+              const { name, minTime, maxTime, labelSets, lastCheck, lastError } = store;
+              const health = lastError ? 'down' : 'up';
+              const color = getColor(health);
+
+              return (
+                <tr key={name}>
+                  <td>{name}</td>
+                  <td>
+                    <Badge color={color}>{health.toUpperCase()}</Badge>
+                  </td>
+                  <td>
+                    <StoreLabels labelSet={labelSets} />
+                  </td>
+                  <td>{minTime >= MAX_TIME ? <FontAwesomeIcon icon={faInfinity} /> : formatTime(minTime)}</td>
+                  <td>{maxTime >= MAX_TIME ? <FontAwesomeIcon icon={faInfinity} /> : formatTime(maxTime)}</td>
+                  <td>
+                    {parseTime(lastCheck) >= MAX_TIME ? (
```

Comment:
I was more thinking about 
```suggestion
                    {parseTime(Math.abs(now() - lastCheck)) >= MAX_TIME ? (
```
To be defensive here :P

Reply:
`lastCheck` is actually in ISO format (I think, it's a string nonetheless). Guess we need to do
```
Math.abs(now() - parseTime(lastCheck))
```
here.